test/e2e/storage/testsuites/disruptive.go

```diff
@@ -25,43 +25,52 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	e2epv "k8s.io/kubernetes/test/e2e/framework/pv"
-	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 	"k8s.io/kubernetes/test/e2e/storage/testpatterns"
 	"k8s.io/kubernetes/test/e2e/storage/utils"
 )
 
 type disruptiveTestSuite struct {
+	TestSuiteInterface
 	tsInfo TestSuiteInfo
 }
 
-var _ TestSuite = &disruptiveTestSuite{}
-
-// InitDisruptiveTestSuite returns subPathTestSuite that implements TestSuite interface
-func InitDisruptiveTestSuite() TestSuite {
-	return &disruptiveTestSuite{
-		tsInfo: TestSuiteInfo{
-			Name:       "disruptive",
-			FeatureTag: "[Disruptive][LinuxOnly]",
-			TestPatterns: []testpatterns.TestPattern{
-				// FSVolMode is already covered in subpath testsuite
-				testpatterns.DefaultFsInlineVolume,
-				testpatterns.FsVolModePreprovisionedPV,
-				testpatterns.FsVolModeDynamicPV,
-				testpatterns.BlockVolModePreprovisionedPV,
-				testpatterns.BlockVolModeDynamicPV,
+// InitCustomDisruptiveTestSuite returns subPathTestSuite that implements TestSuite interface
+// using custom test patterns
+func InitCustomDisruptiveTestSuite(patterns []testpatterns.TestPattern) TestSuiteHandler {
+	return TestSuiteHandler{
+		testSuite: &disruptiveTestSuite{
+			tsInfo: TestSuiteInfo{
+				Name:         "disruptive",
+				FeatureTag:   "[Disruptive][LinuxOnly]",
+				TestPatterns: patterns,
 			},
 		},
 	}
 }
-func (s *disruptiveTestSuite) GetTestSuiteInfo() TestSuiteInfo {
+
+// InitDisruptiveTestSuite returns subPathTestSuite that implements TestSuite interface
+// using test suite default patterns
+func InitDisruptiveTestSuite() TestSuiteHandler {
+	testPatterns := []testpatterns.TestPattern{
+		// FSVolMode is already covered in subpath testsuite
+		testpatterns.DefaultFsInlineVolume,
+		testpatterns.FsVolModePreprovisionedPV,
+		testpatterns.FsVolModeDynamicPV,
+		testpatterns.BlockVolModePreprovisionedPV,
+		testpatterns.BlockVolModeDynamicPV,
+	}
+	return InitCustomDisruptiveTestSuite(testPatterns)
+}
+
+func (s *disruptiveTestSuite) getTestSuiteInfo() TestSuiteInfo {
 	return s.tsInfo
 }
 
-func (s *disruptiveTestSuite) SkipRedundantSuite(driver TestDriver, pattern testpatterns.TestPattern) {
+func (s *disruptiveTestSuite) skipUnsupportedTests(driver TestDriver, pattern testpatterns.TestPattern) {
```

Comment:
why only testpatterns.PreprovisionedPV is selected here?

Reply:
I dont have all the context here. But in this change I am not introducing anything new. It is pure code refactor. So testpatterns.PreprovisionedPV was already selected here before my change.

So by reading at the code it seems once the driver supports dynamic provisioning, then it will skip PreprovisionedPV because the dynamic provisioning test will just cover that.

Reply:
Correct, this is to reduce the number of test combinations. Since this test case doesn't depend on dynamic vs preprovisioned behavior, if the driver supports dynamic provisioning, we'll only test that pattern, and not the preprovisioned pattern.